docs/faq.rst

```diff
@@ -0,0 +1,218 @@
+
+The supported operating systems are listed in the :doc:`rocm:compatibility/compatibility-matrix`
+
+CUDA and OpenCL
+===============
+
+Is HIP a drop-in replacement for CUDA?
+--------------------------------------
+
+The `HIPIFY <https://github.com/ROCm/HIPIFY>`_ tools can automatically convert
+almost all runtime code. Most device code needs no additional conversion since
+HIP and CUDA have similar names for math and built-in functions. HIP code
+provides the same performance as native CUDA code, plus the benefits of running
+on AMD platforms.
+
+Additional porting may only be required to deal with architecture feature
+queries or with CUDA capabilities that HIP doesn't support.
+
+How does HIP compare with OpenCL?
+---------------------------------
+
+HIP offers several benefits over OpenCL:
+
+* Device code can be written in modern C++, including templates, lambdas, classes and so on.
+* Host and device code can be mixed in the source files.
+* The HIP API is less verbose than OpenCL and is familiar to CUDA developers.
+* Porting from CUDA to HIP is significantly easier than porting from CUDA to OpenCL.
+* HIP uses the best available development tools on each platform: ``amdclang++`` for AMD GPUs or ``nvcc``
+  for NVIDIA GPUs, and profilers like ``omniperf`` or ``Nsight Systems``.
+* HIP provides pointers and host-side pointer arithmetic.
+* HIP provides device-level control over memory allocation and placement.
+* HIP offers an offline compilation model.
+
+How does porting CUDA to HIP compare to porting CUDA to OpenCL?
+---------------------------------------------------------------
+
+OpenCL is fairly different from HIP and CUDA when considering the host runtime,
+but even more so when it comes to kernel code.
+The device code of HIP is a C++ dialect, while OpenCL is C99-based.
+OpenCL also does not support single-source compilation.
+
+As a result, the OpenCL syntax is a lot different from HIP, and porting tools
+have to perform complex transformations, especially when it comes to templates
+or other C++ features in kernels.
+
+Can I install both CUDA and ROCm on the same machine?
+-----------------------------------------------------
+
+Yes. Beware, that you still need a compatible GPU to actually run the compiled code.
+
+HIP detected my platform (``amd`` vs ``nvidia``) incorrectly. What should I do?
+-------------------------------------------------------------------------------
+
+If HIP is unable to detect the correct platform, or you want to manually specify
+it, set the ``HIP_PLATFORM`` environment variable to either ``amd`` or ``nvidia``.
+
+On NVIDIA platforms, can I mix HIP code with CUDA code?
+-------------------------------------------------------
+
+Yes. Most HIP data structures are typedefs to CUDA equivalents and can be
+intermixed.
+
+See the :doc:`how-to/hip_porting_guide` for more details.
+
+Compiler related questions
+==========================
+
+How to use HIP-Clang to build HIP programs?
```

Comment:
Please link this tutorial: https://rocm.docs.amd.com/projects/HIP/en/latest/tutorial/saxpy.html#compiling-on-the-command-line

It is written to represent the compiler usage on different OS or platform.

Reply:
I was thinking of whether to even keep this question in the FAQ. That tutorial is a bit overkill for this question. Maybe something similar can also be added to the HIP compilers page of #3622 and we could link to that